api/handlers.go

```diff
@@ -52,6 +70,29 @@ func CreateSessionHandlerFunc(sessions Sessions) http.HandlerFunc {
 
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusCreated)
-		w.Write(sessJson)
+		w.Write(sessJSON)
+	}
+}
+
+func GetSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		ID := getVarsFunc(r)["ID"]
+
+		log.Event(ctx, fmt.Sprintf("get session by ID: %s", ID), log.INFO)
+
+		result, err := cache.Get(ID)
+		if err != nil {
```

Comment:
This needs to return the appropriate status code. If the session is not found 404, otherwise I think a 500 as something went wrong server side.

Reply:
already added the 404 when session isn't found. not sure how to read into the errors to see what type they could be. Is there some sort of switch function I need on it?

Reply:
In Go its sometimes common to have a public error constant for specific cases i.e. `io.EOF` this enables you to check for specific error cases and allows you to handle them how you like.

At the moment I don't think this is necessary here - if the session is not found we will return `nil, nil`. Its the caller's responsibility to check if the session returned is nil - this should be stated in the Godoc for the `Set()` method making it clear what the expectation is.

It is possible this could change - we might encounter a case where a not found error would be a better design. If that happens we can reassess but for now return nil.

Reply:
> At the moment I don't think this is necessary here - if the session is not found we will return nil, nil. Its the caller's responsibility to check if the session returned is nil - this should be stated in the Godoc for the **Set()** method making it clear what the expectation is.

I'm assuming that `Set()` should be the `GetByID()` function instead as thats the one returning the session and the error?

Reply:
Yes sorry my mistake. However I think `Set()` should also return an error - It will be using Redis if the client cannot connect or an incorrect status is returned or it dies midway through the request then we would need to propagate that back through the app and treat the request to the sessions API and unsuccessful.

Reply:
Added error return on Set this morning, was thinking exactly as you said.